commons-packet/commons-packet-service/src/main/java/io/mosip/commons/packetmanager/service/PacketReaderService.java

```diff
@@ -161,6 +163,60 @@ public String getSource(String id, String source, String process) {
         return source;
     }
 
+    public SourceProcessDto getSourceAndProcess(String id, String field, String source, String process) {
+        SourceProcessDto sourceProcessDto = null;
+        if (StringUtils.isEmpty(source)) {
+            try {
+                if (defaultStrategy.equalsIgnoreCase(DefaultStrategy.DEFAULT_PRIORITY.getValue())) {
+                    InfoResponseDto infoResponseDto = info(id);
+                    ContainerInfoDto containerInfoDto = findPriority(field, process, infoResponseDto);
+                    if (containerInfoDto == null)
+                        throw new SourceNotPresentException();
+                    sourceProcessDto = new SourceProcessDto(containerInfoDto.getSource(), containerInfoDto.getProcess());
+                }
```

Comment:
Not required to handle else for this if condition ?

Reply:
it supports only default priority right now, there are 2 more priorities discussed which were not implemented yet, so when we implement other priority strategy then else block will come. Right now always DEFAULT_PRIORITY by default in our config.

Reply:
we should throw a exception if there any spell mistake in the config, but for now it is ok